rule/merge.go

```diff
@@ -41,14 +43,14 @@ import (
 // marked with a "# keep" comment, values in the attribute not marked with
 // a "# keep" comment will be dropped. If the attribute is empty afterward,
 // it will be deleted.
-func MergeRules(src, dst *Rule, mergeable map[string]bool, filename string) {
+func MergeRules(src, dst *Rule, isMergeable, isManaged IsMergeable, filename string) {
```

Comment:
I agree that this is a bug to be fixed, but this changes the signature of a public function and is thus a breaking change. 

Since a `map[string]bool` actually track three states per entry (absent, present with false, present with true), we could potentially use that information to encode the `isManaged` bit in  a mostly backwards compatible way (ignoring that someone may already set a value to `false` explicitly, which would be very unidiomatic).

Reply:
Is this public API? I've never understood what is public API vs just public so other packages within gazelle can use it. Should I consider anything listed on https://pkg.go.dev/github.com/bazelbuild/bazel-gazelle as public?

I can either do the `map[string]bool` `true|false` thing, or maybe create a new method and the old one can keep the current behaviour (while simply invoking the new one).

Do you have any preferences?

Reply:
I've updated with this map `false` value idea, see the fixup commit.